torch/csrc/jit/autodiff.cpp

```diff
@@ -395,6 +397,48 @@ static std::vector<Value*> gradientForNode(Node* node, ArrayRef<Value*> grad_val
       });
       return {backward_value->node()->output(0), nullptr, nullptr, nullptr, nullptr, nullptr};
 
+    } else if (node->matches("aten::thnn_conv2d_forward(Tensor self, Tensor weight, int[] kernel_size, Tensor? bias, int[] stride, int[] padding) -> (Tensor, Tensor, Tensor)")) {
+      auto graph = node->owningGraph();
+
+      auto convNode = graph->create(aten::thnn_conv2d_backward, 3);
+      convNode->addInput(grads.at(0).value());
+      convNode->addInput(inputs.at(0));
+      convNode->addInput(inputs.at(1));
+      convNode->addInput(node->namedInput(attr::kernel_size));
+      convNode->addInput(node->namedInput(attr::stride));
+      convNode->addInput(node->namedInput(attr::padding));
+      convNode->addInput(outputs.at(1));
+      convNode->addInput(outputs.at(2));
+      convNode->addInput(graph->insertConstant(std::vector<bool>{true, true, true}));
+      graph->insertNode(convNode);
+      auto outputs = convNode->outputs();
+      JIT_ASSERT(outputs.size() == size_t(3));
+
+      return {outputs[0], outputs[1], nullptr, outputs[2], nullptr, nullptr};
+
+    } else if (node->matches("aten::native_batch_norm(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps) -> (Tensor, Tensor, Tensor)")) {
+      auto graph = node->owningGraph();
+      auto bnNode = graph->create(aten::native_batch_norm_backward, 3);
+      auto grad_out = grads.at(0);
+      bnNode->addInput(grad_out);
+      bnNode->addInput(inputs.at(0));
+      bnNode->addInput(inputs.at(1));
+      bnNode->addInput(inputs.at(3));
+      bnNode->addInput(inputs.at(4));
+      bnNode->addInput(outputs.at(1));
+      bnNode->addInput(outputs.at(2));
+      bnNode->addInput(inputs.at(5));
+      bnNode->addInput(inputs.at(7));
+      bnNode->addInput(graph->insertConstant(std::vector<bool>{true, true, true}));
+      graph->insertNode(bnNode);
+      auto outputs = fmap<SymbolicVariable>(bnNode->outputs());
```

Comment:
using `graph->insert` would auto insert the outputs here as well, making the code simpler.